### PR TITLE
AKS: add --fqdn-subdomain for private cluster with custom private dns zone

### PR DESCRIPTION
**Pull request by levimm**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Review by gossion — APPROVED** [review]

**Review by feiskyer — APPROVED** [review]
LGTM, please squash the commits

**Review by miwithro — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -976,6 +976,69 @@ def test_aks_create_none_private_dns_zone(self, resource_group, resource_group_l
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
 
+    # add live_only for role assignment GUID
```
Comment (fengzhou-msft) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:979]: You can mock `_gen_guid` in role assignment: https://github.com/Azure/azure-cli-extensions/blob/fea5dcc42918ec56de7477854508b78e83c0da61/src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py#L93-L96

**Review by fengzhou-msft — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (levimm) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py]: wow, thank for the info! Changed to used mock.

**Review by fengzhou-msft — APPROVED** [review]

**Comment by feiskyer** [comment]
@fengzhou-msft  the release pipeline is failing https://github.com/Azure/azure-cli-extensions/runs/2010422413. Are you fixing it?

**Comment by fengzhou-msft** [comment]
It's blocked by a previous version release. Will retry after previous version is released.

**Comment by feiskyer** [comment]
@fengzhou-msft thanks!

**Pull request merged**